pkg/apis/discovery/types.go

```diff
@@ -0,0 +1,139 @@
+	// more than 100.
+	// +listType=set
+	Addresses []string
+	// conditions contains information about the current status of the endpoint.
+	Conditions EndpointConditions
+	// hostname of this endpoint. This field may be used by consumers of
+	// endpoints to distinguish endpoints from each other (e.g. in DNS names).
+	// Multiple endpoints which use the same hostname should be considered
+	// fungible (e.g. multiple A values in DNS). Must pass DNS Label (RFC 1123)
+	// validation.
+	// +optional
+	Hostname *string
+	// targetRef is a reference to a Kubernetes object that represents this
+	// endpoint.
+	// +optional
+	TargetRef *api.ObjectReference
+	// topology contains arbitrary topology information associated with the
+	// endpoint. These key/value pairs must conform with the label format.
+	// https://kubernetes.io/docs/concepts/overview/working-with-objects/labels
+	// Topology may include a maximum of 16 key/value pairs. This includes, but
+	// is not limited to the following well known keys:
+	// * kubernetes.io/hostname: the value indicates the hostname of the node
+	//   where the endpoint is located. This should match the corresponding
+	//   node label.
+	// * topology.kubernetes.io/zone: the value indicates the zone where the
+	//   endpoint is located. This should match the corresponding node label.
+	// * topology.kubernetes.io/region: the value indicates the region where the
+	//   endpoint is located. This should match the corresponding node label.
+	// +optional
+	Topology map[string]string
+}
+
+// EndpointConditions represents the current condition of an endpoint.
+type EndpointConditions struct {
+	// ready indicates that this endpoint is prepared to receive traffic,
+	// according to whatever system is managing the endpoint. A nil value
+	// indicates an unknown state. In most cases consumers should interpret this
+	// unknown state as ready.
+	Ready *bool
+}
+
+// EndpointPort represents a Port used by an EndpointSlice
+type EndpointPort struct {
+	// The name of this port. All ports in an EndpointSlice must have a unique
+	// name. If the EndpointSlice is dervied from a Kubernetes service, this
+	// corresponds to the Service.ports[].name.
+	// Name must either be an empty string or pass IANA_SVC_NAME validation:
+	// * must be no more than 15 characters long
+	// * may contain only [-a-z0-9]
+	// * must contain at least one letter [a-z]
+	// * it must not start or end with a hyphen, nor contain adjacent hyphens
+	Name *string
+	// The IP protocol for this port.
+	// Must be UDP, TCP, or SCTP.
+	Protocol *api.Protocol
+	// The port number of the endpoint.
+	// If this is not specified, ports are not restricted and must be
+	// interpreted in the context of the specific consumer.
+	Port *int32
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// EndpointSliceList represents a list of endpoint slices.
+type EndpointSliceList struct {
```

Comment:
What is the purpose of this type (why is it defined as an api type?)

Reply:
A list type is required for API resources.